jaeger/jaeger.go

```diff
@@ -0,0 +1,141 @@
+	jaegerConf.Float64Var(&samplerParam, "sampler-param", 1, "the sampler parameter (number)")
+	jaegerConf.StringVar(&samplerManagerAddr, "sampler-manager-addr", "http://jaeger:5778/sampling", "The HTTP endpoint when using the remote sampler")
+	jaegerConf.IntVar(&samplerMaxOperations, "sampler-max-operations", 0, "The maximum number of operations that the sampler will keep track of")
+	jaegerConf.DurationVar(&samplerRefreshInterval, "sampler-refresh-interval", time.Second*10, "How often the remotely controlled sampler will poll jaeger-agent for the appropriate sampling strategy")
+	jaegerConf.IntVar(&reporterMaxQueueSize, "reporter-max-queue-size", 0, "The reporter's maximum queue size")
+	jaegerConf.DurationVar(&reporterFlushInterval, "reporter-flush-interval", time.Second*10, "The reporter's flush interval")
+	jaegerConf.BoolVar(&reporterLogSpans, "reporter-log-spans", false, "Whether the reporter should also log the spans")
+	jaegerConf.StringVar(&collectorAddr, "collector-addr", "", "HTTP endpoint for sending spans directly to a collector, i.e. http://jaeger-collector:14268/api/traces")
+	jaegerConf.StringVar(&collectorUser, "collector-user", "", "Username to send as part of 'Basic' authentication to the collector endpoint")
+	jaegerConf.StringVar(&collectorPassword, "collector-password", "", "Password to send as part of 'Basic' authentication to the collector endpoint")
+	jaegerConf.StringVar(&agentAddr, "agent-addr", "Localhost:6831", "UDP address of the agent to send spans to. (only used if collector-endpoint is empty)")
+	globalconf.Register("jaeger", jaegerConf, flag.ExitOnError)
+}
+
+func ConfigProcess() {
+	var err error
+	addTagsParsed, err = parseTags(addTagsRaw)
+	if err != nil {
+		log.Fatalf("jaeger: Config validation error. %s", err)
+	}
+}
+
+// parseTags parses the given string into a slice of opentracing.Tag.
+// the string musti be a comma separated list of key=value pairs, where value
+// can be specified as ${key:default}, where `key` is an environment
+// variable and `default` is the value to use in case the env var is not set
+func parseTags(input string) ([]opentracing.Tag, error) {
+	pairs := strings.Split(input, ",")
+	var tags []opentracing.Tag
+	for _, pair := range pairs {
+		if pair == "" {
+			continue
+		}
+
+		if !strings.Contains(pair, "=") {
+			return nil, fmt.Errorf("invalid tag specifier %q", pair)
+		}
+		kv := strings.SplitN(pair, "=", 2)
+		key, val := strings.TrimSpace(kv[0]), strings.TrimSpace(kv[1])
+		if len(key) == 0 || len(val) == 0 {
+			return nil, fmt.Errorf("invalid tag specifier %q", pair)
+		}
+
+		if strings.HasPrefix(val, "${") && strings.HasSuffix(val, "}") {
+			spec := strings.SplitN(val[2:len(val)-1], ":", 2)
+			envVar := spec[0]
+			var envDefault string
+			if len(spec) == 2 {
+				envDefault = spec[1]
+			}
+			val = os.Getenv(envVar)
+			if val == "" && envDefault != "" {
+				val = envDefault
+			}
+		}
+
+		tags = append(tags, opentracing.Tag{Key: key, Value: val})
```

Comment:
`val` could potentially be `""` if an env var has been used which is not defined, and there's no default. what happens if `val` is `""`? does it just show up like this in jaeger, or would it be better to not add this tag then?

Reply:
not sure, but i checked the jaeger-go-client lib and they do it this way also